File: 0036-valid-sudoku/0036-valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        unordered_map <string, int> umap;
        for(int i=0; i<9; i++) {
            for(int j=0; j<9; j++) {
                if(board[i][j] == '.') continue;

                char ch = board[i][j];
                
                string key1 = string("r") + to_string(i) + to_string(ch);
                string key2 = string("c") + to_string(j) + to_string(ch);
                string key3 = string("b") + to_string(i/3) + to_string(j/3) + to_string(ch);


                if(umap[key1] or umap[key2] or umap[key3]) return false;

                umap[key1]=1;
                umap[key2]=1;
                umap[key3]=1;
            }
        }

        return true;
    }
};

File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        rows = [set() for _ in range (9)]
        cols = [set() for _ in range (9)]
        subs = [set() for _ in range (9)]

        for r in range(9):
            for c in range(9):
                if board[r][c] == ".":
                    pass
                else:
                    if board[r][c] in rows[r]:
                        return False
                    else:
                        rows[r].add(board[r][c])

                    if board[r][c] in cols[c]:
                        return False
                    else:
                        cols[c].add(board[r][c])

                    if board[r][c] in subs[(r//3)*3+c//3]:
                        return False
                    else:
                        subs[(r//3)*3+c//3].add(board[r][c])

        return True       